### PR TITLE
SE-3001:Initial Add for Valuation Analysis Notebook

### PR DESCRIPTION
**Pull request by armlowe**
# Pull Request Checklist

- [x] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [ ] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

New notebook to demonstrate analysis methods for valuations

### CODE REVIEW

File: examples/use-cases/valuation/Valuation Analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,1093 @@
+{
```
Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <ul><li>Third paragraph, sentence should read: 'We'll look at things through the lense of an alternative asset portfolio: 'GlobalAlternatives', as well as a fixed income portfolio: 'GlobalCredit'.</li><li>Next sentence should read: 'These portfolios...' and mention the credit portfolio.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <ul><li>Probably can rename the Excel file to 'valuation_analysis_data.xlsx'</li><li>For the client internals, we should provide a consistent prefix (i.e. cid_) rather than luid_</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Should refer somewhere to the SRS KB article: https://support.lusid.com/knowledgebase/article/KA-01893/en-us</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <ul><li>Can remove 'First' in the sentence 'First, in order to...'.</li><li>Typo in 'somme'.</li><li>Your last sentence is a bit confusing. Maybe reword it a bit more explicitly with the names of the two recipes and say which one allows partial results and which one doesn't.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Can we recommit this file with the cells populated</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <ul><li>Should add uppercase 'for' in second sentence.</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Probably can remove 'Now' from this sentence.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Should refer to manifest KB somewhere: https://support.lusid.com/knowledgebase/article/KA-01892/en-us</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Probably use something other than 'Now...' again. Perhaps 'We then...'</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='46'/>

File: examples/use-cases/valuation/Valuation Analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,4389 @@
...
```
Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Can you parse out the error detail and display rather than the JSON?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>We don't need to display all of these cells as we've already shown the AI values above.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>For the valuation output columns, can we display more user friendly column names rather than the address key</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (jyaffe) [reply examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Can you actually handle this better so that if the error message contains 'DataMap exists', don't print an error.</p><p>e.g.</p><pre class="ql-syntax" spellcheck="false">except lusid.ApiException as e:
    if 'DataMap exists' not in json.loads(e.body)["detail"]:
        print("Error loading data map: " + json.loads(e.body)["detail"])
</pre>

File: examples/use-cases/valuation/Valuation Analysis.ipynb [hunk 3]
```diff
@@ -0,0 +1,3001 @@
...
```
Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.    transactions
</pre><p>would add <code>valuation manifest</code> here </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <blockquote>if a large quote lookback quote window is specified of say 10 days</blockquote><p>remove one of the <code>quote</code>s</p><p> </p><p> </p><p>also</p><blockquote>that can help us gain relavent insights to address</blockquote><p><code>relevant</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #26.    pd.options.display.float_format = '{:,.6f}'.format
</pre><p>do you really want 6 decimal places on every DataFrame? Some of them look excessive. May be better to set it to default at 2, and then explicitly set to 6 on individual DFs that require additional precision.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>It can be a little messy in places but the LUID stands for <code>LUSID Instrument ID</code> and not <code>LUSID internal ID</code>   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # Create two different recipes doepending on the AllowPartiallySuccessfulEvaluation option
</pre><p>typo -&gt; <code>depending</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    def get_valuation(date, portfolio_code, recipe_code, metricsDict):    
</pre><p>the <code>metricsDict</code> parameter should be <code>metrics_dict</code></p><p> </p><p>and same goes for <code>metrics_list</code> and <code>metrics_cols</code> within the function definition  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.            metricsList = []
</pre><p>update <code>metrics_list</code> etc here too. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='42'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>In the text sections you use both<code>Lusid</code> and<code>LUSID</code>	, please can you stick to <code>LUSID</code>  throughout? In the code it is less important because it will depend on the context. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Comment (jyaffe) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #13.    valuation = get_valuation("2022-01-28T00:00:00Z", "GlobalCredit", "ValuationAnalysisRecipeWPartialResult", metrics)#["Aggregation/Errors"][0]
</pre><p>Can you remove the commented out code?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='53'/>

Comment (johnd83) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>First mention in notebook of the SRS. Perhaps link to a KB article for anyone unfamiliar with the SRS?</p><p>e.g. <a href="https://support.lusid.com/knowledgebase/article/KA-01893/en-us" target="_blank">https://support.lusid.com/knowledgebase/article/KA-01893/en-us</a></p><p> </p><p><em>Edit: Ah, I see this link is used later in the notebook. Perhaps link here too/instead as it's the first mention?</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (johnd83) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Typo:</p><blockquote>retreive </blockquote><p>retrieve</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

File: examples/use-cases/valuation/Valuation Analysis.ipynb [hunk 4]
```diff
@@ -0,0 +1,2976 @@
...
```
Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Typo in second retrieved instance.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Comment (Tomwp) [inline examples/use-cases/valuation/Valuation Analysis.ipynb:1]: <p>Would rephrase: "LUSID can't fully model and calculate analytics for at present". </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/finbourne/sample-notebooks/pull/650/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

**Review by Tomwp — APPROVED** [review]

**Comment by Tomwp** [comment]
Approved.
